ArduPlane/GCS_Mavlink.cpp

```diff
@@ -794,13 +794,169 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
+{
+  switch(packet.command) {
+    
+#if OFFBOARD_GUIDED == ENABLED
+    case MAV_CMD_GUIDED_CHANGE_SPEED: {
+        // command is only valid in guided mode
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+         // only airspeed commands are supported right now...
+        if (int(packet.param1) != SPEED_TYPE_AIRSPEED) {  // since SPEED_TYPE is int in range 0-1 and packet.param1 is a *float* this works.
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject airspeeds that are outside of the tuning envelope
+        if (packet.param2 > plane.aparm.airspeed_max || packet.param2 < plane.aparm.airspeed_min) {
+            return MAV_RESULT_UNSUPPORTED;
+        }
+
+         // reject duplicate airspeeds
+        float new_target_airspeed_cm = packet.param2 * 100;
+//        if (new_target_airspeed_cm == plane.guided_state.target_airspeed_cm) {
+        if ( fabsf(new_target_airspeed_cm - plane.guided_state.target_airspeed_cm)<0.001f) { // compare if two floats are 'the same'
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+        plane.guided_state.target_airspeed_cm = new_target_airspeed_cm;
+        plane.guided_state.target_airspeed_time_ms = AP_HAL::millis();
+
+         if (is_zero(packet.param3)) {
+            // the user wanted /maximum acceleration, pick a large value as close enough
+            plane.guided_state.target_airspeed_accel = 1000.0f;
+        } else {
+            plane.guided_state.target_airspeed_accel = fabsf(packet.param3);
+        }
+
+         // assign an acceleration direction
+        if (plane.guided_state.target_airspeed_cm < plane.target_airspeed_cm) {
+            plane.guided_state.target_airspeed_accel *= -1.0f;
+        }
+        return MAV_RESULT_ACCEPTED;
+    }
+
+     case MAV_CMD_GUIDED_CHANGE_ALTITUDE: {
+        // command is only valid in guided
+        if (plane.control_mode != &plane.mode_guided) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+
+        // command is only valid if alt is not at it's default value of -1. this is a float, so the use of is_zero(x-1)  
+        if (  is_zero(packet.z-1) ) {
+            return MAV_RESULT_TEMPORARILY_REJECTED;
+        }
+         // the requested alt data might be relative or absolute
+        float new_target_alt = packet.z * 100;
+        float new_target_alt_rel = packet.z * 100 + plane.home.alt;
+
+         // only global/relative/terrain frames are supported
+        switch(packet.frame) {
+            case MAV_FRAME_GLOBAL_RELATIVE_ALT: {
+                if   (fabsf(plane.guided_state.target_alt - new_target_alt_rel)<0.001f ) { // compare two floats as near-enough
+                    // reject duplicate altitude change requests
+                    return MAV_RESULT_TEMPORARILY_REJECTED;
+                }
+                plane.guided_state.target_alt = packet.z * 100 + plane.home.alt;
```

Comment:
why re-calc this here when new_target_alt_rel is calculated above?

Reply:
yep, ok, willfix.